blockstore/apps/search/fieldsets.py

```diff
@@ -0,0 +1,119 @@
+"""
+FieldSets for search app.
+"""
+from elasticsearch_dsl import (
+    Keyword,
+    InnerDoc as ESInnerDoc,
+    Integer,
+    Text,
+)
+
+from rest_framework import serializers
+from .serializers import FieldSetSerializer
+
+
+class InnerDoc(ESInnerDoc):
+    pass
+
+
+class FieldSet:
+    """
+    A FieldSet describes a collection of ElasticSearch fields which can be
+    dynamically associated with an Index to add support for indexing metadata
+    from different sources related to entities in that Index.
+    """
+
+    name = None  # A string used to namespace the fields.
+    Document = None  # An ElasticSearch InnerDoc.
+    Serializer = None  # A DRF Serializer which can be use to serialize and deserialize the FieldSet.
+
+
+class SummaryFieldSet(FieldSet):
+    """
+    FieldSet for summary data of the entity.
+    """
+
+    name = 'summary'
+
+    class Document(InnerDoc):
+
+        title = Text()
+        description = Text()
+        image = Text(index=False)
```

Comment:
To be clear, this is supposed to be a URL to the image and not the binary content and it is set to just be stored and not be searchable (`index=False` does that). Unlike authors and orgs for which we will likely have database entities, the image may be in the XML or could be a video still etc so yup the idea here is to push the image URL in the ES entity so that it can be returned along with the results. The view layer can use signed URLs if required as well.

In general it is normal to have non-searchable data saved along with the entities in ES.